src/server/worker/master.go

```diff
@@ -249,6 +249,16 @@ func (a *APIServer) spoutSpawner(pachClient *client.APIClient) error {
 	}
 	puller := filesync.NewPuller()
 
+	if a.pipelineInfo.Spout.Marker != "" {
+		markerBranch, err := pachClient.InspectBranch(a.pipelineInfo.Pipeline.Name, "marker")
+		if err != nil || markerBranch == nil {
+			err = pachClient.CreateBranch(a.pipelineInfo.Pipeline.Name, "marker", "", nil)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
Is there a good reason to create this here rather than implicitly the first time a file is put to the branch? I feel like it would probably make the code a little more robust to create it implicitly since there's only two states you can be in, no branch yet before it's run, and having a branch with the marker in it.